### PR TITLE
weekend challenge: rock, paper, scissors, lizard, spock

### PR DESCRIPTION
**Pull request by loushark**
Functionality:
[x] the player can enter their name before the game
[x] the player will be presented the choices (rock, paper, scissors, lizard and spock)
[x] the player can choose one option
[x] the game (computer) will choose a random option
[x] the player will know if they have won or lost

Functionality not added:
[ ] the game is not a two player (the player currently plays against the computer)

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+class RockPaperScissors < Sinatra::Base
+
+  get '/' do
+    erb :home
+  end
+
+  post '/play' do
+    player = Player.new(params[:player_name])
+    @game = Game.create(player)
+    redirect '/play'
```
Comment (Emmapr123) [inline app.rb:14]: Doesn't the computer get confused with these having the same name? I didn't know this was possible! Got me a bit confused when reading it at first, but looks cool.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  def player_chooses(choice)
+    @player_choice = choice
+  end
+
+  def self.random
+    ["Rock", "Paper", "Scissors", "Lizard", "Spock"].sample
+  end
+
+  def win_lose_draw(player, computer)
+    rules = { "Rock" => { "Rock" => "draw", "Paper" => "lose", "Scissors" => "win", "Lizard" => "win", "Spock" => "lose" },
```
Comment (Emmapr123) [inline lib/game.rb:31]: nice use of a hash!

File: spec/features/feature_spec.rb [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    scenario 'the hompage has the title rock paper, scissors' do
+      visit '/'
+      expect(page).to have_content "Rock - Paper - Scissors - Lizard - Spock"
+    end
+  end
+
+  feature 'player can enter a name' do
+    scenario 'the player can see their name on the play page' do
+      sign_in_and_play
+      expect(page).to have_content "Hi Sneaky Racoon!\nLet's play!"
```
Comment (Emmapr123) [inline spec/features/feature_spec.rb:14]: Loving the names again!

File: spec/features/feature_spec.rb [hunk 4]
```diff
@@ -0,0 +1,161 @@
+require 'spec_helper'
+
```
Comment (Emmapr123) [inline spec/features/feature_spec.rb:2]: Very well and thoroughly tested!

File: views/home.erb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<style>
```
Comment (Emmapr123) [inline views/home.erb:1]: I am very curious to see the app now! That styling sounds really lovely

File: views/play.erb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            border: none;
+            border-radius: 50%;
+            margin: 20px
+          }
+</style>
+
+<h1>Hi <%= @game.player.name %>!</h1>
+<h2>Let's play! </h2>
+
+<form action="/result" name="choice" method="post">
```
Comment (Emmapr123) [inline views/play.erb:24]: great form!

File: views/result.erb [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+<% elsif @game.computer_choice == "Scissors" %>
+<h2>Computer played Scissors!</h2>
+<% elsif @game.computer_choice == "Lizard" %>
+<h2>Computer played Lizard!</h2>
+<% elsif @game.computer_choice == "Spock" %>
+<h2>Computer played Spock!</h2>
+<% end %>
+
+<% if @win_lose_draw == "win" %>
+<h1>You win! (づ｡◕‿‿◕｡)づ</h1>
```
Comment (Emmapr123) [inline views/result.erb:46]: Love the drawings! Nice addition

File: README.md [hunk 8]
```diff
@@ -1,85 +1,41 @@
...
-As usual please start by
+#### Functionality:
 
-* Forking this repo
-* TEST driving development of your app
+[x] the player can enter their name before the game  
+[x] the player will be presented the choices (rock, paper, scissors, lizard and spock)  
+[x] the player can choose one option  
+[x] the game (computer) will choose a random option  
+[x] the player will know if they have won or lost
```
Comment (willspencer16) [inline README.md:22]: Great use of a README!

File: spec/features/web_helpers.rb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+def sign_in_and_play
+  visit '/'
+  fill_in 'player_name', with: "Sneaky Racoon"
+  click_button 'Submit'
+end
```
Comment (willspencer16) [inline spec/features/web_helpers.rb:5]: Nice web helper!

File: spec/game_spec.rb [hunk 10]
```diff
@@ -0,0 +1,41 @@
+require 'game'
+
+describe Game do
+  let(:new_game) { Game.new("Crazy Teaspoon") }
+  let(:player1) { "Rock" }
+  let(:player2) { "Paper" }
+  let(:player3) { "Scissors" }
+  let(:computer1) { "Rock" }
+  let(:computer2) { "Paper" }
+  let(:computer3) { "Scissors" }
```
Comment (willspencer16) [inline spec/game_spec.rb:10]: Stubs on point!

File: views/home.erb [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+          }
+</style>
+
+<h1>Rock - Paper - Scissors - Lizard - Spock</h1>
+<h3> what shall we call you? </h3>
+<form action="/play" method="post">
+  <input type="text" name="player_name">
+  <br>
+  <input id="submit" type="submit" value="Submit">
+</form>
```
Comment (willspencer16) [inline views/home.erb:26]: This is what I wanted to do! Great work

File: app.rb (hunk 1 above)
Comment (loushark) [reply app.rb:14]: great catch, i could call this @new_game instead to help the readability

**Pull request closed without merging**